common/djangoapps/student/roles.py

```diff
@@ -153,7 +153,8 @@ def add_users(self, *users):
         # legit get updated.
         from student.models import CourseAccessRole
         for user in users:
-            if user.is_authenticated and user.is_active and not self.has_user(user):
+            user = User.objects.get(id=user.id)
```

Comment:
^ That would be better. Though if you're doing that, you might as well break the rules and import the model directly in the migration (from django.contrib.auth.models import User as RealUser) and then call these unmodified roles methods with that RealUser. It has exactly the same effect, but keeps the roles.py code clean.

If modifying roles.py seems like the only way, we need to add (as comments) a clear explanation in the code, or else someone might come along and "fix" it in the future. (since `User.objects.filter(id__in=(user.id for user in users))` is ugly and looks like an inefficient no-op).